File: c/grains/grains.c
#include "grains.h"
#include <math.h>

uint64_t square(uint8_t index) {
    const uint8_t index_min = 1;
    const uint8_t index_max = 64;

    if (index < index_min || index > index_max) {
        return (uint64_t)0;
    }

    return (uint64_t)pow(2, index - 1);
}

uint64_t total() { return (uint64_t)pow(2, 64); }

File: lua/grains/grains.lua
local grains = {}

function grains.square(index)
    local index_min = 1
    local index_max = 64

    if index < index_min or index > index_max then
        return 0
    end

    return 2 ^ (index - 1)
end

function grains.total()
    return 2 ^ 64
end

return grains
